### ISSUE TITLE
managements_treatments: missing managements and/or treatments

### ISSUE THREAD

**Issue by gsrohde** [PecanProject/bety#171]
MISSING MANAGEMENTS:

select management_id, count(*), array_agg(treatment_id) as treatment_id_list from managements_treatments where management_id not in (select id from managements) group by management_id;
 management_id | count | treatment_id_list 
---------------+-------+-------------------
          1696 |     1 | {735}
          1703 |     1 | {735}
          2313 |     1 | {746}
          2315 |     1 | {740}
          2317 |     1 | {741}
          2401 |     1 | {1}
          2402 |     1 | {1}
(7 rows)

MISSING TREATMENTS:

select treatment_id, count(*), array_agg(management_id) as management_id_list from managements_treatments where treatment_id not in (select id from treatments) group by treatment_id;
 treatment_id | count |                    management_id_list                    
--------------+-------+----------------------------------------------------------
          141 |     4 | {818,1852,1851,817}
          229 |     4 | {819,823,1853,1857}
          232 |     6 | {1859,1855,1856,822,821,825}
          233 |     6 | {1854,826,822,820,1856,1860}
          689 |    11 | {1661,1677,1676,1675,1674,1666,1665,1664,1663,1662,1660}
          690 |    11 | {1668,1660,1661,1662,1663,1667,1669,1674,1675,1676,1677}
          691 |     8 | {1676,1674,1663,1660,1662,1661,1677,1675}
          692 |     9 | {1660,1670,1661,1677,1676,1675,1674,1662,1663}
          693 |     9 | {1662,1660,1677,1676,1675,1674,1671,1663,1661}
          694 |     9 | {1672,1674,1663,1662,1661,1660,1675,1676,1677}
          695 |     9 | {1673,1677,1676,1663,1662,1661,1660,1675,1674}
          698 |     1 | {2400}
          842 |     1 | {1832}
          849 |     9 | {1660,1662,1661,1677,1676,1675,1674,1834,1663}
         1094 |     1 | {2370}
         1146 |     1 | {2370}
(16 rows)

**Comment by dlebauer** [comment]
A check of the citations with missing managements shows that these orphans have been replaced by new managements

```sql select * from managements where id in (select management_id from managements_treatments where treatment_id not in (select id from treatments))

select * from managements where citation_id = 151 and id in (select management_id from managements_treatments where treatment_id not in (select id from treatments))
select * from managements where citation_id = 151 and id not in (select management_id from managements_treatments where treatment_id not in (select id from treatments))

select * from managements where citation_id = 64 and id in (select management_id from managements_treatments where treatment_id not in (select id from treatments))
select * from managements where citation_id = 64 and id not in (select management_id from managements_treatments where treatment_id not in (select id from treatments))
```

So I have deleted these

```sql

delete from managements_treatments where management_id not in (select id from managements)
delete from managements_treatments where treatment_id not in (select id from treatments)
```

**Comment by gsrohde** [comment]
Are there managements_treatments entries for the new managements that correspond to the old managements?

**Comment by dlebauer** [comment]
> Are there managements_treatments entries for the new managements that correspond to the old managements?

Yes - I found them by 1) Identifying the associated citation and then 2) querying managements associated with the citation. This is related to a glitch in the interface that prevents data entry technicians from unlinking managements.

**Comment by gsrohde** [comment]
Is there a bug report for that?

**Comment by dlebauer** [comment]
Bug report for inability to unlink managements is here: https://ebi-forecast.igb.illinois.edu/redmine/issues/2334

Indeed, it looks like the feature is implemented in pull request #106, but is still awaiting tests to be written.
